### DISCUSSION TITLE
怎么监听到群里面谁改了昵称

### DISCUSSION THREAD

**Discussion by yujumpjump** [mamoe/mirai#2527] [Q&A]
mirai 有没有监听群成员改了昵称的世件呢

**Comment by NoMathExpectation** [comment]
https://docs.mirai.mamoe.net/EventList.html

**Reply by yujumpjump** [reply]
大佬。谢谢你。不过我还有一个问题。就是如何让bot去监听一个给定的群。比如使用注解啊 让一个方法事件去监听给定的群

**Reply by NoMathExpectation** [reply]
https://docs.mirai.mamoe.net/Events.html#%E8%BF%87%E6%BB%A4

**Reply by yujumpjump** [reply]
请问 MemberCardChangeEvent 这个事件可不可以立刻监听到群成员修改了名片，

**Reply by NoMathExpectation** [reply]
是的，你可以自己试一下

**Reply by yujumpjump** [reply]
我看了 好像不是

**Reply by yujumpjump** [reply]
成员群名片改动. 此事件广播前修改就已经完成.
由于服务器并不会告知名片变动, 此事件只能由 mirai 在发现变动时才广播. 不要依赖于这个事件

**Reply by yujumpjump** [reply]
他告诉我的是这个

**Reply by NoMathExpectation** [reply]
那应该就这样了，也应该没其他办法立即知道了

**Reply by yujumpjump** [reply]
大佬 我还有一个问题 就是onlineAnnouncement.members  这个方法 是不是只能返回50个人啊 我看不管参数给的是false 还是true 都是给我50个

**Reply by NoMathExpectation** [reply]
这个我就不清楚了，建议自己去查一下文档或者代码注释，也有可能是个bug

**Reply by yujumpjump** [reply]
如果是bug 在哪可以反馈呢

**Reply by NoMathExpectation** [reply]
去开新issue
